server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteStoreMigrationAllocationIT.java

```diff
@@ -0,0 +1,929 @@
+            setClusterMode(MIXED_MODE);
+        }
+
+        prepareIndex(1, 0, getRemoteStoreEnabledIndexSettingsBuilder(isRemoteStoreBackedIndex));
+        assertEquals(
+            (isRemoteStoreBackedIndex ? "true" : null),
+            client.admin()
+                .cluster()
+                .prepareState()
+                .execute()
+                .actionGet()
+                .getState()
+                .getMetadata()
+                .index(TEST_INDEX)
+                .getSettings()
+                .get(SETTING_REMOTE_STORE_ENABLED)
+        );
+
+        Decision decision = getDecisionForTargetNode(nonRemoteNode, true, true, false);
+        Decision.Type type = Decision.Type.NO;
+        String reason = "[remote_store migration_direction]: primary shard copy can not be allocated to a non-remote node";
+        if (isMixedMode == false) {
+            // strict mode
+            type = Decision.Type.YES;
+            reason =
+                "[remote_store migration_direction]: primary shard copy can be allocated to a non-remote node for strict compatibility mode";
+        } else if (isRemoteStoreBackedIndex) {
+            reason =
+                "[remote_store migration_direction]: primary shard copy can not be allocated to a non-remote node because a remote store backed index's shard copy can only be allocated to a remote node";
+        }
+        assertEquals(type, decision.type());
+        assertEquals(reason, decision.getExplanation().toLowerCase(Locale.ROOT));
+
+        logger.info(" --> attempt allocation");
+        client.admin()
+            .indices()
+            .prepareUpdateSettings(TEST_INDEX)
+            .setSettings(
+                Settings.builder()
+                    .put("index.routing.allocation.include._name", nonRemoteNodeName)
+                    .put("index.routing.allocation.exclude._name", allNodesExcept(nonRemoteNodeName))
+            )
+            .execute()
+            .actionGet();
+
+        if (isMixedMode == false) {
+            ensureGreen(TEST_INDEX);
+
+            logger.info(" --> verify allocation of primary shard");
+            RoutingTable routingTable = client.admin().cluster().prepareState().execute().actionGet().getState().getRoutingTable();
+            ShardRouting primaryShardRouting = routingTable.index(TEST_INDEX).shard(0).primaryShard();
+            assertAllocation(primaryShardRouting, nonRemoteNode);
+        } else {
+            ensureRed(TEST_INDEX);
+
+            logger.info(" --> verify non-allocation of primary shard");
+            RoutingTable routingTable = client.admin().cluster().prepareState().execute().actionGet().getState().getRoutingTable();
+            ShardRouting primaryShardRouting = routingTable.index(TEST_INDEX).shard(0).primaryShard();
+            assertNonAllocation(primaryShardRouting);
+        }
+    }
+
+    public void testAllocateNewPrimaryShardOnRemoteNodeForRemoteStoreDirection() throws Exception {
```

Comment:
Should we add an non-remote node as well in the cluster and make sure the index is not assigned to it?